skunkworks_crow/src/main/java/org/odk/share/views/ui/instance/adapter/InstanceAdapter.java

```diff
@@ -54,14 +58,47 @@ public void onClick(View v) {
             }
         });
         holder.title.setText(cursor.getString(cursor.getColumnIndex(InstanceProviderAPI.InstanceColumns.DISPLAY_NAME)));
-        holder.subtitle.setText(cursor.getString(cursor.getColumnIndex(InstanceProviderAPI.InstanceColumns.DISPLAY_SUBTEXT)));
+
+        long lastStatusChangeDate = getCursor().getLong(getCursor().getColumnIndex(InstanceProviderAPI.InstanceColumns.LAST_STATUS_CHANGE_DATE));
+        String status = getCursor().getString(getCursor().getColumnIndex(InstanceProviderAPI.InstanceColumns.STATUS));
+        String subtext = getDisplaySubtext(context, status, new Date(lastStatusChangeDate));
+
+        holder.subtitle.setText(subtext);
         long id = cursor.getLong(cursor.getColumnIndex(InstanceProviderAPI.InstanceColumns._ID));
         holder.checkBox.setChecked(selectedInstances.contains(id));
         holder.reviewedForms.setVisibility(View.GONE);
         holder.unReviewedForms.setVisibility(View.GONE);
 
     }
 
+    public static String getDisplaySubtext(Context context, String state, Date date) {
+        try {
+            if (state == null) {
```

Comment:
Can those `if-else` replaced by `switch-case` sentences?

Reply:
I don't want these to be changed because I've copied these things exactly from ODK-Collect code so that if it gets changed in future then we would be able to identify things easily.